### ISSUE TITLE
Claude3 Support

### ISSUE THREAD

**Issue by zhlmmc** [promptfoo/promptfoo#523]
Claude3 is released, seems like they changed the API:
![image](https://github.com/promptfoo/promptfoo/assets/1726527/a08e8a97-11fb-4ad1-8286-72c6578988d2)

**Comment by typpo** [comment]
Wondering if @streichsbaer, who wrote our mistral provider, has any interest in this one :)

I'd love to make this change but am in the hospital with a family member right now.  Otherwise, will probably get to it in the coming weeks.

**Comment by aantn** [comment]
@typpo hope all is well soon!

**Comment by streichsbaer** [comment]
Funny you should say that. I just came here to offer my help with this and saw the message :)

On it!

**Comment by streichsbaer** [comment]
Alrighty, have just opened a PR that adds support for Claude 3.
@typpo, let me know if there is anything else I can help with 👍 .

@zhlmmc, you can test the changes locally by following these steps:
1. Clone the repo
2. Check out the PR branch
3. Add the new provider to your `promptfooconfig.yaml`
```
provider:
- id: anthropic:messages:claude-3-opus-20240229
    config:
      temperature: 0.0
```
4. You can use the OpenAI prompt template, e.g. `prompts: [prompt.json]`. System prompts are optional.
```
[
  {
    "role": "system",
    "content": "{{ system_message }}"
  },
  {
    "role": "user",
    "content": "{{ question }}"
  }
]
```
5. Then just run `npx /path/to/promptfoo eval`

Let me know if there are any issues!

**Comment by typpo** [comment]
Thanks @streichsbaer!  This is released in 0.45.0
